cpp/arcticdb/storage/s3/s3_storage.cpp

```diff
@@ -32,6 +34,48 @@ std::streamsize S3StreamBuffer::xsputn(const char_type* s, std::streamsize n) {
 }
 }
 
+void check_ec2_metadata_endpoint(const S3ApiInstance& s3_api) {
+    auto disabled_env = Aws::Environment::GetEnv("AWS_EC2_METADATA_DISABLED");
+    if (Aws::Utils::StringUtils::ToLower(disabled_env.c_str()) == "true") {
+        const char* who_disabled = s3_api.is_ec2_metadata_disabled_by_us() ?
+            "EC2 metadata endpoint did not respond in time so was bypassed":
+            "AWS_EC2_METADATA_DISABLED environment variable is set to true";
+        log::storage().warn("{}. This means machine identity authentication will not work.", who_disabled);
+    }
+}
+
+bool S3Storage::check_creds_and_bucket() {
+    using namespace Aws::S3;
+    // We cannot easily change the timeouts of the s3_client_, so use async:
+    auto future = s3_client_.HeadBucketCallable(Model::HeadBucketRequest().WithBucket(bucket_name_.c_str()));
+    auto wait = std::chrono::milliseconds(ConfigsMap::instance()->get_int("S3Storage.CheckBucketMaxWait", 1000));
+    if (future.wait_for(wait) == std::future_status::ready) {
+        auto outcome = future.get();
+        if (!outcome.IsSuccess()) {
```

Comment:
```
if (outcome.IsSuccess()) {
  return true;
}
// log some stuff
return false;
```

is clearer IMO. There's only one way we can return true here and the code could make that more obvious.